datafusion/expr/src/expr.rs

```diff
@@ -1692,6 +1692,24 @@ fn create_names(exprs: &[Expr]) -> Result<String> {
         .join(", "))
 }
 
+/// Whether the given expression is volatile, i.e. whether it can return different results
```

Comment:
Volatility is already captured as part of the function signature. I think you can access it like `func.volatility()`:

https://github.com/apache/arrow-datafusion/blob/7f312c8a1d82b19f03c640e4a5d7d6437b2ee835/datafusion/expr/src/built_in_function.rs#L360-L363

Reply:
Oh, cool. I tried to find one but didn't find it.